### PR TITLE
fix: added support for ga4 v2 hybrid mode

### PR DESCRIPTION
**Pull request by utsabc**
## What are the changes introduced in this PR?

Added support for Hybrid mode in Ga4 V2 by adding functionalities to read specific integration object values based on integration type and updating `clientId` and `sessionId` for the measurement protocol events

## What is the related Linear task?

Resolves INT-2444

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

File: src/v0/destinations/ga4/utils.js [hunk 1]
```diff
@@ -474,11 +474,21 @@ const basicValidation = (event) => {
  * @param {*} message
  * @returns
  */
-const getGA4ClientId = (message, Config) => {
+const getGA4ClientId = (message, Config, destType) => {
   let clientId;
 
   if (isHybridModeEnabled(Config)) {
-    const integrationsObj = getIntegrationsObj(message, 'ga4');
+    let integrationsObj = {};
```
Comment (krishna2020) [inline src/v0/destinations/ga4/utils.js:481]:
```suggestion
    const integrationsObj = getIntegrationsObj(message, destType);
```
won't this suffice ?

File: test/integrations/destinations/ga4_v2/processor/customMappings.ts [hunk 2]
```diff
@@ -540,6 +540,7 @@ export const customMappingTestCases = [
               },
               body: {
                 JSON: {
```
Comment (krishna2020) [inline test/integrations/destinations/ga4_v2/processor/customMappings.ts:542]: can we have a test case for getting `sessionId` & `sessionNumber` also to be included ?

Comment (utsabc) [reply test/integrations/destinations/ga4_v2/processor/customMappings.ts:563]: Added

File: test/integrations/testUtils.ts [hunk 3]
```diff
@@ -25,7 +25,9 @@ export const getTestDataFilePaths = (dirPath: string, opts: OptionValues): strin
   const globPattern = join(dirPath, '**', 'data.ts');
   let testFilePaths = globSync(globPattern);
   if (opts.destination) {
-    testFilePaths = testFilePaths.filter((testFile) => testFile.includes(opts.destination));
+    testFilePaths = testFilePaths.filter((testFile) =>
+      testFile.includes(`/destinations/${opts.destination}/`),
```
Comment (krishna2020) [inline test/integrations/testUtils.ts:29]: why this directory update ?

Comment (utsabc) [reply test/integrations/testUtils.ts:29]: When running test cases by giving the destination flag
`includes` matches both ga4 and ga4_v2 for `ga4`

Hence fixed it

**Review by krishna2020 — APPROVED** [review]

**Review by koladilip — APPROVED** [review]

**Pull request merged**